libc/stdio_scanf.c

```diff
@@ -0,0 +1,3701 @@
+	TEST_ASSERT_EQUAL_STRING("tak", chr);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(tested_str, "%% %s", chr));
+	TEST_ASSERT_EQUAL_STRING("tak", chr);
+}
+
+TEST(stdio_scanf_str_chr_ptr, str_diff)
+{
+
+	char tested_str[50];
+	char chr[50];
+	int val_hex;
+	int val;
+	char spc[] = "This is something what 123 needed 0x0005";
+
+	sprintf(tested_str, "%s", spc);
+	create_file(std_path, tested_str);
+	rd = fopen(std_path, "r");
+
+	TEST_ASSERT_EQUAL_INT(7, vfscand_wrapper(rd, "%s %s %s %s %d %s %x", chr, chr, chr, chr, &val, chr, &val_hex));
+	TEST_ASSERT_EQUAL_STRING("needed", chr);
+	TEST_ASSERT_EQUAL_HEX(0x0005, val_hex);
+	TEST_ASSERT_EQUAL_INT(123, val);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(7, fscanf(rd, "%s %s %s %s %d %s %x", chr, chr, chr, chr, &val, chr, &val_hex));
+	TEST_ASSERT_EQUAL_STRING("needed", chr);
+	TEST_ASSERT_EQUAL_HEX(0x0005, val_hex);
+	TEST_ASSERT_EQUAL_INT(123, val);
+
+	TEST_ASSERT_EQUAL_INT(7,
+		vsscanf_wrapper(tested_str, "%s %s %s %s %d %s %x", chr,
+			chr, chr, chr, &val, chr, &val_hex));
+	TEST_ASSERT_EQUAL_STRING("needed", chr);
+	TEST_ASSERT_EQUAL_HEX(0x0005, val_hex);
+	TEST_ASSERT_EQUAL_INT(123, val);
+
+	TEST_ASSERT_EQUAL_INT(7, sscanf(tested_str, "%s %s %s %s %d %s %x", chr, chr, chr, chr, &val, chr, &val_hex));
+	TEST_ASSERT_EQUAL_STRING("needed", chr);
+	TEST_ASSERT_EQUAL_HEX(0x0005, val_hex);
+	TEST_ASSERT_EQUAL_INT(123, val);
+}
+
+TEST(stdio_scanf_str_chr_ptr, str_no_spaces)
+{
+
+	char tested_str[50];
+	char chr[50];
+	int val_hex;
+	char spc[] = "somethingishappeneing123rightnow/0x0005";
+
+	sprintf(tested_str, "%s", spc);
+	create_file(std_path, tested_str);
+	rd = fopen(std_path, "r");
+
+	TEST_ASSERT_EQUAL_INT(2, vfscand_wrapper(rd, "%[^0x]0x%x", chr, &val_hex));
+	TEST_ASSERT_EQUAL_HEX(0x0005, val_hex);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(2, fscanf(rd, "%[^0x]0x%x", chr, &val_hex));
+	TEST_ASSERT_EQUAL_HEX(0x0005, val_hex);
+
+	TEST_ASSERT_EQUAL_INT(
+		2, vsscanf_wrapper(tested_str, "%[^0x]0x%x", chr, &val_hex));
+	TEST_ASSERT_EQUAL_HEX(0x0005, val_hex);
+
+	TEST_ASSERT_EQUAL_INT(2, sscanf(tested_str, "%[^0x]0x%x", chr, &val_hex));
+	TEST_ASSERT_EQUAL_HEX(0x0005, val_hex);
+}
+
+TEST(stdio_scanf_str_chr_ptr, str_white_signs)
+{
+
+	char tested_str[50];
+	char chr[50];
+	char spc[] = "stdio fscanf test/0x0005";
+
+	sprintf(tested_str, "%s", spc);
+	create_file(std_path, tested_str);
+	rd = fopen(std_path, "r");
+
+	TEST_ASSERT_EQUAL_INT(1, vfscand_wrapper(rd, "%[^\n]%*c", chr));
+	TEST_ASSERT_EQUAL_STRING(spc, chr);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(1, fscanf(rd, "%[^\n]%*c", chr));
+	TEST_ASSERT_EQUAL_STRING(spc, chr);
+
+	TEST_ASSERT_EQUAL_INT(1, vsscanf_wrapper(tested_str, "%[^\n]%*c", chr));
+	TEST_ASSERT_EQUAL_STRING(spc, chr);
+
+	TEST_ASSERT_EQUAL_INT(1, sscanf(tested_str, "%[^\n]%*c", chr));
+	TEST_ASSERT_EQUAL_STRING(spc, chr);
+}
+
+TEST(stdio_scanf_str_chr_ptr, pointers)
+{
+
+	char tested_str[50];
+	void **chr;
+	char spc[] = "0x12ab34cd";
```

Comment:
Please use one of _hexspeak_ values to identify that this is particularly not a valid address.